sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubClientBuilder.java

```diff
@@ -267,6 +270,34 @@ public EventHubClientBuilder configuration(Configuration configuration) {
         return this;
     }
 
+    /**
+     * Sets a custom endpoint address when connecting to the Event Hubs service. This can be useful when your network
+     * does not allow connecting to the standard Azure Event Hubs endpoint address, but does allow connecting through
+     * an intermediary. For example: {@literal https://my.custom.endpoint.com:55300}.
+     * <p>
+     * If no port is specified, the default port for the {@link #transportType(AmqpTransportType) transport type} is
+     * used.
+     *
+     * @param customEndpointAddress The custom endpoint address.
+     * @return The updated {@link EventHubClientBuilder} object.
+     * @throws IllegalArgumentException if {@code customEndpointAddress} cannot be parsed into a valid {@link URL}.
+     */
+    public EventHubClientBuilder customEndpointAddress(String customEndpointAddress) {
```

Comment:
The HTTP client builders just use `endpoint` not `endpointAddress`. So, should we use similar terminology here? `customEndpoint()`?

Reply:
https://github.com/Azure/azure-sdk/issues/1367
https://gist.github.com/jsquire/f0948c9a5ffe2741b03d312dcf8d899e#gistcomment-3345232

Looks like the board decided on customEndpointAddress